R/ranger.R

```diff
@@ -715,6 +723,52 @@ ranger <- function(formula = NULL, data = NULL, num.trees = 500, mtry = NULL,
   }
   
   class(result) <- "ranger"
+  
+  ## Prepare quantile prediction
+  if (quantreg) {
+    terminal.nodes <- predict(result, data, type = "terminalNodes")$predictions
+    n <- result$num.samples
+    result$random.node.values <- matrix(nrow = max(terminal.nodes), ncol = num.trees)
+    
+    ## Select one random obs per node and tree
+    for (tree in 1:num.trees){
+      idx <- sample(1:n, n)
+      result$random.node.values[terminal.nodes[idx, tree], tree] <- response[idx]
+    }
+    
+    ## Prepare out-of-bag quantile regression
+    if(!is.null(result$inbag.counts)) {
+      inbag.counts <- simplify2array(result$inbag.counts)
+      random.node.values.oob <- 0 * terminal.nodes
+      random.node.values.oob[inbag.counts > 0] <- NA
+      
+      ## For each tree and observation select one random obs in the same node (not the same obs)
+      for (tree in 1:num.trees){
+        is.oob <- inbag.counts[, tree] == 0
+        num.oob <- sum(is.oob)
+        
+        if (num.oob != 0) {
+          oob.obs <- which(is.oob)
+          oob.nodes <- terminal.nodes[oob.obs, tree]
+          for (j in 1:num.oob) {
+            random.node.values.oob[oob.obs[j], tree] <- sample(response[terminal.nodes[-oob.obs[j], tree] == oob.nodes[j]], size = 1)
```

Comment:
@mnwright I have the same problem in my patch of OOB quantile predictions (pointed out in the following issue https://github.com/lorismichel/quantregForest/issues/4). Due to the fact that the function `sample` does change behavior when a numeric of length one is provided as first argument (and `size=1`).

Reply:
Thanks. I already noticed that, see commit 66d943173ad696b94bb3c7620e565c4f5fa27c28. 
Actually, I wanted to open a PR in quantregForest but figured you already did ;)

Reply:
great! (sorry I miss this commit)